src/FakeItEasy/Creation/DefaultFakeAndDummyManager.cs

```diff
@@ -48,19 +49,20 @@ public bool TryCreateDummy(Type typeOfDummy, out object result)
             return false;
         }
 
-        private static IFakeOptions CreateFakeOptions(Type typeOfFake, ProxyOptions proxyOptions)
+        private static IFakeOptions CreateFakeOptions<T>(ProxyOptions proxyOptions) => new FakeOptions<T>(proxyOptions);
+
+        private static Func<ProxyOptions, IFakeOptions> GetFakeOptionsFactory(Type typeOfFake)
         {
-            var optionsConstructor = typeof(FakeOptions<>)
-                .MakeGenericType(typeOfFake)
-                .GetConstructor(new[] { typeof(ProxyOptions) });
+            var method = typeof(DefaultFakeAndDummyManager).GetMethod(nameof(CreateFakeOptions), BindingFlags.NonPublic | BindingFlags.Static).MakeGenericMethod(typeOfFake);
 
-            return (IFakeOptions)optionsConstructor.Invoke(new object[] { proxyOptions });
+            return (Func<ProxyOptions, IFakeOptions>)method.CreateDelegate(typeof(Func<ProxyOptions, IFakeOptions>));

```

Comment:
Nice! It's annoying that you can't create a delegate directly from a constructor... In my test I had created the delegate from a Linq expression, but your approach is more efficient.

Reply:
Oh. I had absolutely no idea what I was doing. I figured I'd done the worst possible thing and you'd fix it for me!